components/raftstore-v2/src/raft/peer.rs

```diff
@@ -115,6 +119,12 @@ pub struct Peer<EK: KvEngine, ER: RaftEngine> {
     pending_messages: Vec<RaftMessage>,
 
     gc_peer_context: GcPeerContext,
+
+    /// Record the instants of peers being added into the configuration.
+    /// Remove them after they are not pending any more.
+    pub peers_start_pending_time: Vec<(u64, Instant)>,
```

Comment:
Should not be public.

How about making them in a struct and then using an `Option<Box>`?

Reply:
Got it, but to clarify do you mean create a struct like 
```
sturct PeersStatus {
    start_pending_time: Vec<u64, Instant>,
    down_ids: Vec<u64>
}
```
And make it Option<Box>?

Reply:
Yes. You can refer to `CompactLogContext` to see how code is organized by functionality.

Reply:
Got it, let me try.